### PR TITLE
SI-7469 Remove misc. @deprecated elements

### PR DESCRIPTION
**Pull request by soc**

### CODE REVIEW

File: src/reflect/scala/reflect/api/Types.scala [hunk 1]
```diff
@@ -878,7 +878,9 @@ trait Types {
    *  until we start tracking explicit kinds equivalent to typeFun (except that the latter requires tparams nonEmpty).
    *  @group Types
    */
-  def polyType(tparams: List[Symbol], tpe: Type): Type
+  def genPolyType(tparams: List[Symbol], tpe: Type): Type
```
Comment (xeno-by) [inline src/reflect/scala/reflect/api/Types.scala]: Why `genPolyType`?

File: src/reflect/scala/reflect/internal/Scopes.scala [hunk 2]
```diff
@@ -387,7 +387,7 @@ trait Scopes extends api.Scopes { self: SymbolTable =>
       if (toList forall p) this
       else newScopeWith(toList filter p: _*)
     )
-    @deprecated("Use `toList.reverse` instead", "2.10.0")
+    @deprecated("Use `toList.reverse` instead", "2.10.0") // Used in SBT 0.12.4
```
Comment (xeno-by) [inline src/reflect/scala/reflect/internal/Scopes.scala:390]: @harrah could you comment on this?

File: src/reflect/scala/reflect/api/Types.scala (hunk 1 above)
Comment (xeno-by) [inline src/reflect/scala/reflect/api/Types.scala]: Oh I see. I suggest we still continue to use the `polyType` name in the public API.

**Comment by xeno-by** [comment]
Thanks for your work!

**Comment by xeno-by** [comment]
I'd also like to note that the commit message is sort of inconsistent with its content.

File: src/reflect/scala/reflect/internal/Scopes.scala (hunk 2 above)
Comment (harrah) [inline src/reflect/scala/reflect/internal/Scopes.scala:390]: What do you want to know?

Comment (xeno-by) [inline src/reflect/scala/reflect/internal/Scopes.scala:390]: Given that SBT depends on these APIs, how do we best organize the completion of the deprecation cycle?

**Comment by adriaanm** [comment]
Final review by @xeno-by once @soc addresses current feedback.

**Comment by soc** [comment]
@xeno-by Updated commit message, removed deprecation from `polyType`.

File: src/reflect/scala/reflect/internal/Scopes.scala (hunk 2 above)
Comment (xeno-by) [inline src/reflect/scala/reflect/internal/Scopes.scala:390]: ping @harrah

**Comment by xeno-by** [comment]
LGTM

File: src/reflect/scala/reflect/internal/Scopes.scala (hunk 2 above)
Comment (harrah) [inline src/reflect/scala/reflect/internal/Scopes.scala:390]: If sbt 0.13 no longer depends on these, then it is a matter of when you want to break users on 0.12, right?

**Comment by adriaanm** [comment]
so long, deprecatees!

**Pull request merged**